src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/mapView.js

```diff
@@ -68,29 +74,131 @@ class MapViewVisualization extends Visualization {
         }
 
         let defaultMapStyle = 'mapbox://styles/mapbox/light-v9';
+        let accentColor     = [1,234,146];
         if (document.getElementById("root").classList.contains("dark")){
             defaultMapStyle = 'mapbox://styles/mapbox/dark-v9';
+            accentColor     = [222,162,47];
         }
 
+        let preparedDataPoints = []
+        let computed           = this.prepareDataPoints(parsedData,preparedDataPoints,accentColor);
+
+        const scatterplotLayer = new deck.ScatterplotLayer({
+            data: preparedDataPoints,
+            getFillColor: d => d.color,
+            getRadius: d => d.radius
+        })
+        //
+        let latitudeMatch  = parsedData.latitude !== undefined && parsedData.latitude !== null;
+        let latitude       = latitudeMatch ? parsedData.latitude : computed.latitude;
+        let longitudeMatch = parsedData.longitude !== undefined && parsedData.longitude !== null;
+        let longitude      = longitudeMatch ? parsedData.longitude : computed.longitude;
+        // TODO : Compute zoom somehow.
+        let zoomMatch      = parsedData.zoom !== undefined && parsedData.zoom !== null;
+        let zoom           = zoomMatch ? parsedData.zoom : computed.zoom;
 
         const deckgl = new deck.DeckGL({
             container: 'map',
-            mapboxApiAccessToken: 'pk.eyJ1IjoiZW5zby1vcmciLCJhIjoiY2tmNnh5MXh2MGlyOTJ5cWdubnFxbXo4ZSJ9.3KdAcCiiXJcSM18nwk09-Q',
+            mapboxApiAccessToken: TOKEN,
             mapStyle: parsedData.mapStyle || defaultMapStyle,
             initialViewState: {
-                longitude: parsedData.longitude || 0.0,
-                latitude: parsedData.latitude || 0.0,
-                zoom: parsedData.zoom || 3,
+                longitude: longitude,
+                latitude: latitude,
+                zoom: zoom,
                 pitch: parsedData.pitch || 0
             },
             controller: parsedData.controller || true
         });
 
         deckgl.setProps({
-            layers: parsedData.layers || []
+            layers: [scatterplotLayer]
+        });
+    }
+
+    /**
+     * Prepares data points to be shown on the map.
+     * @param preparedDataPoints - List holding geoPoints.
+     * @param parsedData - All the parsed data to create GeoPoints from.
+     * @param accentColor - accent color of IDE if element doesn't specify one.
+     */
+    prepareDataPoints(parsedData, preparedDataPoints, accentColor) {
+        let latitude  = 0.0;
+        let longitude = 0.0;
+        let zoom      = 11;
+
+        if (parsedData.type === GEO_POINT) {
+            this.pushGeoPoint(preparedDataPoints,parsedData,accentColor);
+            latitude  = parsedData.latitude;
+            longitude = parsedData.longitude;
+        } else if (Array.isArray(parsedData) && parsedData.length && parsedData[0].type === GEO_POINT) {
```

Comment:
what is `parsedData[0].type === GEO_POINT` for? Please extract it to well named variable to indicate what it is really checking semantically.